modules/extension/complex/src/main/java/org/geotools/data/complex/expression/FeaturePropertyAccessorFactory.java

```diff
@@ -260,14 +262,8 @@ public boolean canHandle(Object object, String xpath, Class target) {
         public <T> T get(Object object, String xpath, Class<T> target)
                 throws IllegalArgumentException {
 
-            JXPathContext context = JXPathContext.newContext(object);
-            Enumeration declaredPrefixes = namespaces.getDeclaredPrefixes();
-            while (declaredPrefixes.hasMoreElements()) {
-                String prefix = (String) declaredPrefixes.nextElement();
-                String uri = namespaces.getURI(prefix);
-                context.registerNamespace(prefix, uri);
-            }
-
+            JXPathContext context =
+                    JXPathUtils.newSafeContext(object, false, this.namespaces, true);
```

Comment:
Minor feedback, `newSafeContext` may promise too much, if attribute security is used for something like GeoFence security plugin this would not know enough to prevent specific data traversals.

A name like `newDataContext` would be more clear this is about providing access to data only, not functions.

Reply:
I named this based on two existing cases for similar types of issues although `getSafeContext` would be a more consistent name:
org.geoserver.template.TemplateUtils.getSafeConfiguration()
org.geotools.ysld.YamlUtil.getSafeYaml()